pkg/crd/testdata/cronjob_types.go

```diff
@@ -43,7 +43,7 @@ import (
 const DefaultRefValue = "defaultRefValue"
 
 // CronJobSpec defines the desired state of CronJob
-// +kubebuilder:validation:XValidation:rule="has(oldSelf.forbiddenInt) || !has(self.forbiddenInt)",message="forbiddenInt is not allowed",fieldPath=".forbiddenInt",reason="FieldValueForbidden"
+// +kubebuilder:validation:XValidation:rule="has(oldSelf.forbiddenInt) || !has(self.forbiddenInt)",message="forbiddenInt is not allowed",fieldPath=".forbiddenInt",reason="FieldValueForbidden",optionalOldSelf=true
```

Comment:
Does the rule need to change to account for the fact that `oldSelf` is now optional?

Reply:
Updated in the latest push to include a new field with an explicit example of `oldSelf` being optional.